### DISCUSSION TITLE
Welcome to horologist Discussions!

### DISCUSSION THREAD

**Discussion by yschimke** [google/horologist#26] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Nags-RH** [comment]
Hi @yschimke 

We are planning to use Horologist library with latest release version of 0.6.17 . Some of the queries we have are

- Can we use horologist library of our production ready app which will be launched in play-store for commercial purpose?

- As we are using 0.6.x , is this a stable release branch?

- Is horologist version tightly coupled with wear OS version [If we upgrade or downgrade will it also affect  the horologist version]?

- We are planning to use 0.6.17 release version ,since the updates are going to happen only in the main branch if we continue to use the same release branch will their be any breakages /discontinuity of the api’s in future?

- It would be kind of you to mention if you are aware of any known production apps (available on the Play Store) that are built with Horologist?

- Will there be any impact in our code if we upgrade or downgrade the horologist version?

Any insights or experiences you can share would be greatly appreciated! Thanks!

**Reply by yschimke** [reply]
> Can we use horologist library of our production ready app which will be launched in play-store for commercial purpose?

Yes. 

> As we are using 0.6.x , is this a stable release branch?

It's using unstable versions of Wear Compose, but 0.6.x will be fairly stable. Those compose dependencies will be RC and stable soon.

0.7.x will be less stable and on new versions.

> Is horologist version tightly coupled with wear OS version [If we upgrade or downgrade will it also affect the horologist version]?

No. Most of it works down to Wear 2, and won't be noticeably different.

> We are planning to use 0.6.17 release version ,since the updates are going to happen only in the main branch if we continue to use the same release branch will their be any breakages /discontinuity of the api’s in future?

We won't break APIs on 0.6.x. if it happens accidentally we can revert.

> It would be kind of you to mention if you are aware of any known production apps (available on the Play Store) that are built with Horologist?

Most of the Google apps. Most other actively maintained apps, at least for things like scrolling away time text. 

> Will there be any impact in our code if we upgrade or downgrade the horologist version?

Not sure I follow what you mean?

**Reply by Nags-RH** [reply]
"Is the horologist library backward compatible?"

Thanks alot for replying to our queries.😊

**Reply by yschimke** [reply]
Generally we don't guarantee binary compatibility, you shouldn't build a library against it that would break if methods changed.

There are some classes that graduated to Wear Compose that we removed as it would be a worse experience to keep using.

But if you are compiling source against it, it should be fairly simple to upgrade.

And ask questions if not.

**Reply by Nags-RH** [reply]
Hi @yschimke had one more question , After we rollout our app to production and there is an issue with horologist, how soon can we expect a fix if we raise the issue?

**Reply by yschimke** [reply]
It depends. Practically, given timezones, it's probably a 3 day turnaround if the bug and fix are obvious and important.

But there may be things we don't prioritise immediately. As it's open source you can submit a PR if it's blocking you.

There can also be things that can't be solved in Horologist and need a fix in Androidx, which is a whole another timeline.

**Reply by Nags-RH** [reply]
Understood. Thanks alot for replying to our queries so swiftly .
